src/res_cli.ml

```diff
@@ -258,19 +270,23 @@ module CliArgProcessor = struct
           else exit 1
         end
         else
+          let parsetree = match ppx with
+            | `Jsx -> Reactjs_jsx_ppx.rewrite_implementation parseResult.parsetree
+            | `None -> parseResult.parsetree
+          in
           printEngine.printImplementation
-            ~width ~filename ~comments:parseResult.comments parseResult.parsetree
+            ~width ~filename ~comments:parseResult.comments parsetree
     with
     | Failure txt ->
       prerr_string txt;
       prerr_newline();
       exit 1
     | _ -> exit 1
-  [@@raises exit]
+  [@@raises Invalid_argument, exit]
```

Comment:
What raises `Invalid_argument`?

Reply:
The existing react ppx uses that exception to note cases with unexpected situations. I can count at least 19 different places, e.g.

https://github.com/jchavarri/syntax/blob/23336db4e0b7d9f524db573b4e94afe510db43ab/src/reactjs_jsx_ppx.ml#L351-L354